packages/api/src/controllers/stream.ts

```diff
@@ -1836,7 +1864,12 @@ app.get("/:id/config", authorizer({ anyAdmin: true }), async (req, res) => {
   let stream = await db.stream.getByPlaybackId(id, {
     useReplica: false,
   });
-  if (!stream || stream.deleted || stream.suspended) {
+  if (
+    !stream ||
+    stream.deleted ||
+    stream.suspended ||
+    stream.projectId !== req.project?.id
```

Comment:
Maybe a general concern you should have with these updates is regarding **admin** APIs. This is an example of such API, which is called internally (maybe by catalyst) with an admin API key which has cross-user access and thus should have cross-project access as well.

This means that we shouldn't have checks like this on most APIs that have
`authorizer({ anyAdmin: true })`
or
`authorizer({ admin: true })` (admin-restricted but only with a JWT (dashboard access))

There are also functions that have special logic in them for admin queries, which is the case of the List APIs where I made some comments about this too.

TLDR is: On this API you should have no `projectId` check. An admin API key should have access to any resource. There are likely other places where this is happening too, so you need to double check the conditions you change to see if they are breaking an admin API or an admin feature within an API.

Reply:
In general, I think the (naive) mindset should be: "if there's a userId check, there should be a projectId check":
- Here, there are no user ID checks, so there wasn't a need for a projectId check either.
- On places where we check userId but allow admins to override it, there should be a project Id check but admin check should also override it.
- On queries where we have conditionals on how to include the `userId` (normally around admin access), we should pay attention to those when including the `projectId` as well.

So it's like the project and user checks should happen on the same "logical level". I'm not sure if this will hold everywhere in the code tho, as I said it's like a naive mindset without actually checking all usages out there =P

Reply:
Thanks for these - helped a lot to clear through some confusion I had.